test/launch-test.ts

```diff
@@ -0,0 +1,64 @@
+/*
+Copyright 2022 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import * as os from "os";
+import * as fs from "fs";
+import * as path from "path";
+import "expect-playwright";
+import { _electron as electron } from "playwright";
+import { ElectronApplication, Page } from "playwright-core";
+
+describe("App launch", () => {
+    const artifactsPath = path.join(process.cwd(), "test_artifacts");
+    fs.mkdirSync(artifactsPath);
+
+    const tmpDir = fs.mkdtempSync(path.join(os.tmpdir(), "element-desktop-tests"));
+    console.log("Using temp profile directory: ", tmpDir);
+
+    let app: ElectronApplication;
+    let window: Page;
+
+    beforeAll(async () => {
+        const args = ["--profile-dir", tmpDir];
+
+        const executablePath = process.env["ELEMENT_DESKTOP_EXECUTABLE"];
+        if (!executablePath) {
+            // Unpackaged mode testing
+            args.unshift("./lib/electron-main.js");
+        }
+
+        app = await electron.launch({
+            executablePath,
+            args,
+            recordVideo: {
+                dir: artifactsPath,
+            }
+        });
+        window = await app.firstWindow();
+    });
+
+    afterAll(async () => {
+        await app?.close();
+        fs.rmSync(tmpDir, { recursive: true });
```

Comment:
Not entirely sure what the likelyhood of `app.close()` throwing but for safety and avoiding leaving stuff behind we should maybe do something like

```js
try {
  await app?.close();
} finally {
  fs.rmSync(tmpDir, { recursive: true });
}
```

What do you think?

Reply:
Probably a sane idea, but if the app doesn't close and we delete its data from under it we can probably make the situation worse